### PR TITLE
Add support for WooCommerce Analytics module in @woocommerce/explat

### LINKED ISSUE #45128: Support Jetpack's WooCommerce Analytics in @woocommerce/explat
Currently, `@woocommerce/explat` uses `wcTracks` to check for a merchant anon cookie to link an experiment to a track event, this is currently only taking into account the admin side and it doesn't support frontend events being triggered by WooCommerce analytics module `wca`.

### PR DESCRIPTION
**Pull request by senadir**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR adds support for WooCommerce Analytics module (wca) in @woocommerce/explat.

WCA is used to trigger track events for shoppers, the underlying technology is the same, as both of them use the same `tk_ai` cookie.

So in this PR, alongside checking for WCTracks Enabled, I also check if wca.push is defined, which is the trigger that confirms if we have permission to track or not.

Closes https://github.com/woocommerce/woocommerce/issues/45128

### How to test the changes in this Pull Request:

This is tricky to test as it requires deploying an experiment and making sure it actually runs.

1. Add `"@woocommerce/tracks": "workspace:^",` as a dependency in `plugins/woocommerce-blocks/package.json`
2. In `plugins/woocommerce-blocks/assets/js/blocks/checkout/inner-blocks/checkout-actions-block/block.tsx` import `{ Experiment }` from `'@woocommerce/explat'`;
3. Create 2 new elements inside your Block component:
```jsx
const placeOrderButton = (
		<PlaceOrderButton
			label="Default Button"
			fullWidth={ ! showReturnToCart }
		/>
	);

	const placeOrderButtonWithExperiment = (
		<PlaceOrderButton
			label="Experiment Button"
			fullWidth={ ! showReturnToCart }
		/>
	);
```
4. Replace the `PlaceOrderButton` call with this:
```jsx
<Experiment
	name="woocommerce_example_place_order_experiment"
	defaultExperience={ placeOrderButton }
	treatmentExperience={ placeOrderButtonWithExperiment }
	loadingExperience={ <div>⏳</div> }
/>
```
5. On a live, jetpack enabled website, visit Checkout.
6. You should not see any errors in console.
7. In network, there should be a request to `https://public-api.wordpress.com/wpcom/v2/experiments/` with a response saying `request not sampled`

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [x] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message
Add support for WooCommerce Analytics.

#### Comment 

</details>

### CODE REVIEW

File: packages/js/explat/src/index.ts [hunk 1]
```diff
@@ -20,11 +20,14 @@ declare global {
...
 		};
+		_wca: {
+			push?: ( cb: () => void ) => void;
+		};
 	}
 }
 
 export const initializeExPlat = (): void => {
-	if ( window.wcTracks?.isEnabled ) {
+	if ( window.wcTracks?.isEnabled || window?._wca?.push !== undefined ) {
```
Comment (wavvves) [inline packages/js/explat/src/index.ts]: This is used in a lot of places, should it justify a helper method? Also would be more readable than negatives and or's or and's 🤔

**Comment by psealock** [comment]
>I also check if wca.push is defined, which is the trigger that confirms if we have permission to track or not.

It would be good to confirm this. `_wca` gets set here, but from memory I'm not sure if the check happens here or elsewhere.

https://github.com/Automattic/jetpack/blob/7be9301593a4912742295a81dca6cdb6070de572/projects/packages/woocommerce-analytics/src/class-woocommerce-analytics.php#L89

Some quick digging reveals this function as the check, but I'm unsure where it gets called

https://github.com/search?q=repo%3AAutomattic%2Fjetpack%20should_track_store&type=code

Sorry for the quick note without much background, I can do some more poking around very soon.

**Comment by senadir** [comment]
@psealock from out previous work it seems that at least, `wca.push` was only set if you have tracking enabled. The `should_track_store` defines if you should initiate the class or not.

File: packages/js/explat/src/index.ts (hunk 1 above)
Comment (wavvves) [reply packages/js/explat/src/index.ts]: Sorry @senadir, my suggestion[ broke tests](https://github.com/woocommerce/woocommerce/actions/runs/8098082820/job/22130723586?pr=45131) on `assignment-test.js` that now need updating. After this tweak, I'm ready to approve it 👍🏼

File: packages/js/explat/src/error.ts [hunk 2]
```diff
@@ -26,7 +26,7 @@ export const logError = (
 		if ( isDevelopmentMode ) {
 			console.error( '[ExPlat] ', error.message, error ); // eslint-disable-line no-console
 		} else {
-			if ( ! window.wcTracks?.isEnabled ) {
+			if ( ! window.wcTracks?.isEnabled && ! window?._wca?.push ) {
```
Comment (psealock) [inline packages/js/explat/src/error.ts:29]: Can we do this? You'll need the `import` too of course.
```suggestion
			if ( ! canTrack ) {
```

**Review by psealock — APPROVED** [review]
This is looking good @senadir, just a small comment, Approving 🚀

**Comment by wavvves** [comment]
Maybe my last comment was collapsed in the helper var suggestion, tests are failing :/ 

>Sorry @senadir, my suggestion[ broke tests](https://github.com/woocommerce/woocommerce/actions/runs/8098082820/job/22130723586?pr=45131) on assignment-test.js that now need updating. After this tweak, I'm ready to approve it 👍🏼

**Review by wavvves — COMMENTED** [review]
(submitting this as a review comment to see if this stops the missing review pings 🙄)

Maybe my last comment was collapsed in the helper var suggestion, tests are failing :/ 

>Sorry @senadir, my suggestion[ broke tests](https://github.com/woocommerce/woocommerce/actions/runs/8098082820/job/22130723586?pr=45131) on assignment-test.js that now need updating. After this tweak, I'm ready to approve it 👍🏼

**Pull request merged**